broker/src/test/java/io/camunda/zeebe/broker/system/partitions/ZeebePartitionTest.java

```diff
@@ -81,6 +82,48 @@ public void shouldInstallLeaderPartition() {
     verify(transition).toLeader(1);
   }
 
+  @Test
+  public void shouldInstallFollowerPartition() {
+    // given
+    schedulerRule.submitActor(partition);
+
+    // when
+    partition.onNewRole(Role.FOLLOWER, 1);
+    schedulerRule.workUntilDone();
+
+    // then
+    verify(transition).toFollower(1);
+  }
+
+  @Test
+  public void shouldUpdateCurrentTermAndRoleAfterTransition() {
+    // given
+    schedulerRule.submitActor(partition);
+    schedulerRule.workUntilDone();
+
+    // when
+    partition.onNewRole(Role.FOLLOWER, 2);
+    schedulerRule.workUntilDone();
+
+    // then
+    verify(ctx, atLeast(1)).setCurrentRole(Role.FOLLOWER);
+    verify(ctx, atLeast(1)).setCurrentTerm(2);
+  }
+
+  @Test
+  public void shouldUseCurrentTermForInactiveTransition() {
+    // given
+    schedulerRule.submitActor(partition);
+    when(ctx.getCurrentTerm()).thenReturn(3L);
+
+    // when -- this internally triggers transition to inactive
+    partition.onSnapshotReplicationStarted();
```

Comment:
:x: Tbh. This looks weird. Could we just call onNewRole with inactive?

Reply:
My thoughts: We bound our self or at least the test to the logic of this method/functionally this makes the test bridle and you have to change a test which is totally unrelated to a feature

Reply:
You are right. That was how I did it first. But I wanted to trigger an internal transition to inactive to ensure that it is using the term from the context. With `onNewRole` we are providing a term, even though for inactive transitions we are not using it.
I will change it to use `onNewRole` as you suggested.